mmengine/infer/infer.py

```diff
@@ -380,11 +380,50 @@ def _load_model_from_metafile(self, model: str) -> Tuple[Config, str]:
                 model_aliases = [alias.lower() for alias in model_aliases]
             if (model_name == model or model in model_aliases):
                 cfg = Config.fromfile(
-                    osp.join(package_path, '.mim', model_cfg['Config']))
+                    join_path(config_dir, model_cfg['Config']))
                 weights = model_cfg['Weights']
                 weights = weights[0] if isinstance(weights, list) else weights
                 return cfg, weights
-        raise ValueError(f'Cannot find model: {model} in {project}')
+        raise ValueError(f'Cannot find model: {model} in {self.scope}')
+
+    @staticmethod
+    def _get_repo_or_mim_dir(scope):
+        """Get the directory where the ``Configs`` located when the package is
+        installed or ``PYTHONPATH`` is set.
+
+        Args:
+            scope (str): The scope of repository.
+
+        Returns:
+            str: The directory of the ``Config``.
+        """
+        try:
+            module = importlib.import_module(scope)
+        except ImportError:
+            if scope not in MODULE2PACKAGE:
+                raise KeyError(
+                    f'{scope} is not a valid scope. The available scopes '
+                    f'are {MODULE2PACKAGE.keys()}')
+            else:
+                project = MODULE2PACKAGE[scope]
+                raise ImportError(
+                    f'Cannot import {scope} correctly, please try to install '
+                    f'the {project} by "pip install {project}"')
+        # Since none of OpenMMLab series packages are namespace packages
+        # (https://docs.python.org/3/glossary.html#term-namespace-package),
+        # The first element of module.__path__ means package installation path.
+        package_path = module.__path__[0]
+
+        if exists(join_path(osp.dirname(package_path), 'configs')):
```

Comment:
`osp.dirname(package_path)` must be a local directory so it is unnecessary to use `join_path`.

Reply:
I think we should only use the functions provided by `fileio`, without having to worry about when to use functions from `os.path` and when to use functions from `fileio`. This uniform programming convention is easier for developing and understanding.